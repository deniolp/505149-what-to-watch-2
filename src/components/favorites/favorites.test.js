import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';

import Favorites from './favorites';
import filmsListMock from '../../mocks/films';

describe(`Favorites`, () => {
  it(`renders correctly`, () => {
    const middleware = [thunk];
    const mockStore = configureMockStore(middleware);
    const initialState = {
      genre: `All genres`,
      films: [],
      promo: {},
      comments: [],
      filmsCounter: 8,
      playingFilm: false,
      isAuthorizationRequired: false,
      user: {},
      favorites: [],
      isReviewSending: false,
      didReviewSend: false,
    };
    const store = mockStore(initialState);

    const tree = renderer.create(
        <BrowserRouter><Provider store={store}><Favorites
          user={{
            id: 1,
            name: `Olga`,
            email: `r@ya.ru`,
            avatarUrl: `/img/photo.jpg`,
          }}
          favorites={filmsListMock.slice(0, 2)}
          match={{
            params: {
              id: 1,
            },
          }}
        /></Provider></BrowserRouter>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
