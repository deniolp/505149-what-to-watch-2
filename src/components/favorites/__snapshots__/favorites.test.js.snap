// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites renders correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href={null}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <a
          href="/favorites"
          onClick={[Function]}
          style={
            Object {
              "color": "#dfcf77",
              "textDecoration": "none",
            }
          }
        >
          <img
            alt="User avatar"
            height="63"
            src="https://htmlacademy-react-2.appspot.com/img/photo.jpg"
            width="63"
          />
        </a>
      </div>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    />
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
